fixtures/bench_binary_search.cpp.sem.uast

```diff
@@ -0,0 +1,2252 @@
+                                                                        offset: 149,
+                                                                        line: 6,
+                                                                        col: 22,
+                                                                     },
+                                                                     end: { '@type': "uast:Position",
+                                                                        offset: 153,
+                                                                        line: 6,
+                                                                        col: 26,
+                                                                     },
+                                                                  },
+                                                                  Name: "high",
+                                                               },
+                                                            },
+                                                            'Prop_Operand1': { '@type': "cpp:CPPASTIdExpression",
+                                                               '@role': [Binary, Expression, Left, Variable],
+                                                               '@pos': { '@type': "uast:Positions",
+                                                                  start: { '@type': "uast:Position",
+                                                                     offset: 143,
+                                                                     line: 6,
+                                                                     col: 16,
+                                                                  },
+                                                                  end: { '@type': "uast:Position",
+                                                                     offset: 146,
+                                                                     line: 6,
+                                                                     col: 19,
+                                                                  },
+                                                               },
+                                                               ExpressionType: "int",
+                                                               ExpressionValueCategory: "LVALUE",
+                                                               IsLValue: true,
+                                                               'Prop_Name': { '@type': "uast:Identifier",
+                                                                  '@pos': { '@type': "uast:Positions",
+                                                                     start: { '@type': "uast:Position",
+                                                                        offset: 143,
+          
```

Comment:
Wow. I wonder what the hell is this :grin: Maybe calling string conversion method on some complex structure?

Reply:
It's really a not so complex function body, but this driver is very verbose (even after several liposuctions in our side).

Reply:
Damn, Github is doing something weird with comments. It should be in a totally different place:
https://github.com/bblfsh/cpp-driver/pull/16/files/3f6092d41862e714e25c5ab2466b244ea8a00ed1#diff-c79cc6306ed8f76da214a1657ea888efR1076

Reply:
Ahhh, the funny catch-all `ExpressionType` field. It's some kind of signature the native AST generates for every expression. For simple ones it's very nice since it gives the inferred type (like `const char[14]`), for others is some beauty like this. When there is a parse error it'll also add some string indicating the error in that field.